### PR TITLE
[KIWI-1378] New Abort Endpoint

### PR DESCRIPTION
**Pull request by ViveakV**
### What changed

New Abort Endpoint

**Before**
image

image

**After**

image

**Second call to Abort**
image

### CODE REVIEW

File: deploy/bav-spec.yaml [hunk 1]
```diff
@@ -582,6 +582,127 @@ paths:
...
+            Content-Security-Policy:
+              schema:
+                type: "string"
+            Feature-Policy:
+              schema:
+                type: "string"
+            Referrer-Policy:
+              schema:
+                type: "string"
+      # TODO should be changed to "both" when lambda is using the body
```
Comment (Caitlin-cooling) [inline deploy/bav-spec.yaml:693]: Do we need this comment?

File: deploy/template.yaml [hunk 2]
```diff
@@ -251,6 +251,7 @@ Globals:
           Fn::ImportValue: !Sub "${L2DynamoStackName}-person-identity-table-name"
         SESSION_TABLE:
           Fn::ImportValue: !Sub "${L2DynamoStackName}-session-table-name"
+        ISSUER: !FindInMap [EnvironmentVariables, !Ref Environment, ISSUER]
```
Comment (Caitlin-cooling) [inline deploy/template.yaml:254]: nice

File: deploy/template.yaml [hunk 3]
```diff
@@ -651,6 +651,182 @@ Resources:
...
+            KeyId:
+              Fn::ImportValue: !Sub "${L2DynamoStackName}-session-table-key-id"
+        - Statement:
+            - Sid: KMSPolicy
+              Effect: Allow
+              Action:
+                - kms:*
+              Resource:
+                "*"
+              #Fn::ImportValue: !Sub "${L2KMSStackName}-vc-signing-key"
```
Comment (Caitlin-cooling) [inline deploy/template.yaml:688]: I know this is something that we have commented out all over the place - do we have plans to implement it at any time? Or shall we just remove?

File: src/services/AbortRequestProcessor.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  	if (!session) {
+  		this.logger.error("No session found for session id", {
+  			messageCode: MessageCodes.SESSION_NOT_FOUND,
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No session found with the session id: ${sessionId}`);
+  	}
+
+  	this.logger.appendKeys({
+  		govuk_signin_journey_id: session?.clientSessionId,
```
Comment (Caitlin-cooling) [inline src/services/AbortRequestProcessor.ts:57]: Do we also want the sessionId?

File: src/services/BavService.ts [hunk 5]
```diff
@@ -37,6 +37,8 @@ export class BavService {
 	}
 
 	async getSessionById(sessionId: string): Promise<ISessionItem | undefined> {
+		console.log(this.tableName);
+		console.log(sessionId);
```
Comment (Caitlin-cooling) [inline src/services/BavService.ts:41]: Pls remove

File: src/tests/api/BavHappyPath.test.ts [hunk 6]
```diff
@@ -158,11 +159,101 @@ describe("BAV CRI: /userinfo Endpoint Happy Path Tests", () => {
...
+		const sqsMessage = await getSqsEventList("txma/", sessionId, 2);
+		await validateTxMAEventData(sqsMessage);
+
+		// Ensure the URI header is present & valid
+		expect(response.headers).toBeTruthy();
+		expect(response.headers.location).toBeTruthy();
+
+		const responseURI = decodeURIComponent(response.headers.location);
+		const responseURIParameters = new URLSearchParams(responseURI);
+		expect(responseURIParameters.has('error')).toBe(true);
```
Comment (Caitlin-cooling) [inline src/tests/api/BavHappyPath.test.ts:187]: @jackmardell looks like there is some linting errors here, mind taking a look?

File: deploy/template.yaml (hunk 3 above)
Comment (ViveakV) [reply deploy/template.yaml]: Removed

File: deploy/bav-spec.yaml (hunk 1 above)
Comment (ViveakV) [reply deploy/bav-spec.yaml]: Removed

**Review by Caitlin-cooling — APPROVED** [review]
Lovely stuff!

**Review by Caitlin-cooling — DISMISSED** [review]

File: src/services/AbortRequestProcessor.ts (hunk 4 above)
Comment (ViveakV) [reply src/services/AbortRequestProcessor.ts:56]: Gonna leave it out only cause we shouldn't be accessing the DB anyways and IF we need to, we can search with the gov ID

**Pull request merged**